metrics/web/test/project_groups/presentation/state/project_groups_notifier_test.dart

```diff
@@ -0,0 +1,874 @@
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+import 'package:metrics/common/domain/entities/persistent_store_error_code.dart';
+import 'package:metrics/common/domain/entities/persistent_store_exception.dart';
+import 'package:metrics/common/presentation/models/persistent_store_error_message.dart';
+import 'package:metrics/common/presentation/models/project_model.dart';
+import 'package:metrics/common/presentation/strings/common_strings.dart';
+import 'package:metrics/project_groups/domain/entities/project_group.dart';
+import 'package:metrics/project_groups/domain/usecases/add_project_group_usecase.dart';
+import 'package:metrics/project_groups/domain/usecases/delete_project_group_usecase.dart';
+import 'package:metrics/project_groups/domain/usecases/parameters/add_project_group_param.dart';
+import 'package:metrics/project_groups/domain/usecases/parameters/delete_project_group_param.dart';
+import 'package:metrics/project_groups/domain/usecases/parameters/update_project_group_param.dart';
+import 'package:metrics/project_groups/domain/usecases/receive_project_group_updates.dart';
+import 'package:metrics/project_groups/domain/usecases/update_project_group_usecase.dart';
+import 'package:metrics/project_groups/presentation/state/project_groups_notifier.dart';
+import 'package:metrics/project_groups/presentation/view_models/project_checkbox_view_model.dart';
+import 'package:metrics/project_groups/presentation/view_models/project_group_card_view_model.dart';
+import 'package:metrics/project_groups/presentation/view_models/project_group_dialog_view_model.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import '../../../test_utils/matcher_util.dart';
+
+void main() {
+  group("ProjectGroupsNotifier", () {
```

Comment:
Can we also test the case when a project group contains a project id of project that was removed? As we don't provide the UI for the projects configuration - I want to make sure that if somebody misconfigured projects - it will continue to function normally.